File: 04/C++/cadastro.cpp
/*
Programa:   cadastro
Arquivo:    cadastro.cpp
*/
#include <iostream>
#include <limits>
#include "cadastro.h"

using namespace std;

Pessoa::Pessoa(){
	}

void Pessoa::cadastrar() {
	int inputSexo;
	cout << "Digite o nome: " ;
	cin >> nome;

	/* Validação de input: Permitir números inteios e
	não aceitar números negativos. 	*/ 
	do {
		cout << "Digite a idade:  " ;
		cin >> idade;
		if(cin.fail()){
			cout << "Digite um número inteiro.\n" ;
			cin.clear();
			cin.ignore(std::numeric_limits<streamsize>::max(), '\n');
			idade = -1; // Forçar permanecer no laço do... while
		}
	} while(idade < 0 );

	/* Validação do Sexo: 1 para Masculino, 2 para Feminino. 			*/
	cout<<"Selecione o sexo: \t 1) Masculino \t 2) Feminino\n";
	cin>> inputSexo;
	switch (inputSexo) {
		case 1:
			sexo = 'M';
		break;
		case 2:
			sexo = 'F';
		break;
		default:
			sexo = 'M';
		break;
	}
}

void Pessoa::mostrar() {
	if (sexo == 'F' && idade >= 30) {
		cout << nome << "\t   não é elegante revelar a idade de uma mulher. "<< endl;
	} else {
		cout << nome << "\t-\t"<< idade << " anos\t-\t"<< sexo << endl;
	}
}

File: 04/Python3/cadastro.py
#!/usr/bin/python3
# coding: utf-8
#
# Programa: cadastro
"""Cadastro de Pessoa"""


class Pessoa(object):
    """ Classe Pessoa: responsável em armazenar dados de uma pessoa. """

    __nome = ""
    __idade = -1
    __sexo = ''

    def __init__(self):
        """ Construtor Python. """

    def cadastra(self):
        """ Método cadastra: permite receber os dados de uma pessoa. """
        self.__nome = input("Digite o seu nome: ")
        while self.__idade < 0:
            try:
                self.__idade = int(input("Digite sua idade: "))
            except ValueError:
                print("\t\tDigite número inteiro !")
        self.__sexo = input("Sexo: M para masculino ou F para feminino: ")
        self.__sexo = self.__sexo.upper()
        if self.__sexo != 'F':
            self.__sexo = 'M'

    def mostra(self):
        """ Método mostra: apresenta os dados cadastrados de uma pessoa. """
        if self.__sexo == 'F' and self.__idade > 30:
            print(self.__nome + '\t idade secreta \t ' + self.__sexo)
        else:
            print(self.__nome + '\t' + str(self.__idade) + '\t' + self.__sexo)

PESSOAS = list()
for i in range(0, 3):
    OBJ = Pessoa()
    OBJ.cadastra()
    PESSOAS.append(OBJ)

for PESSOA in PESSOAS:
    PESSOA.mostra()
"""print ('Linha 46 ' + str(PESSOAS))
print ('Linha 47 ' + str(PESSOAS[0]))
print ('Linha 48 ' + str(PESSOAS[0].__dict__))
print ('Linha 49 ' + str(PESSOAS[0].__dict__.keys()))
print ('Linha 50 ' + str(PESSOAS[0].__dict__.values()))
PESSOAS[0]._Pessoa__idade = 'BlaBlaBla'
print ('Linha 52  ' + str(PESSOAS[0]._Pessoa__idade))
PESSOAS[0]._Pessoa__apelido = 'Cabeção'
print ('Linha 54 ' + str(PESSOAS[0].__dict__))

"""